torch/onnx/_type_utils.py

```diff
@@ -87,11 +88,15 @@ def from_name(
     ) -> JitScalarType:
         """Convert a JIT scalar type or torch type name to ScalarType.
 
+        NB: DO NOT USE this API if dtype is comes from a torch._C.Value.type().scalarType() call because
+            "RuntimeError: INTERNAL ASSERT FAILED at "../aten/src/ATen/core/jit_type_base.h" can
```

Comment:
Curious if there is anything we can do to fix/improve the `INTERNAL ASSERT FAILED` as well?

Reply:
The assert screams at dereferencing inexistent attributes. It could return none, for instance, but changing behavior this low in the stack is quite intrusive